packages/flutter/lib/src/animation/tween.dart

```diff
@@ -256,6 +256,35 @@ class Tween<T extends dynamic> extends Animatable<T> {
   T lerp(double t) {
     assert(begin != null);
     assert(end != null);
+    assert(() {
+      // Assertions that attempt to catch common cases of tweening types
+      // that do not conform to the Tween requirements.
+      dynamic? result;
+      try {
+        result = begin + (end - begin) * t;
+        result as T;
+        return true;
+      } on NoSuchMethodError {
+        throw FlutterError.fromParts(<DiagnosticsNode>[
+          ErrorSummary('Cannot tween between $begin and $end.'),
+          ErrorDescription(
+            'The type ${begin.runtimeType} does not fully implement `+`, `-`, and/or `*`.',
+          ),
+          if (begin is Color || end is Color)
```

Comment:
maybe `T == Color` ?

Reply:
we'd have to check `Color`/`MaterialColor` and others, whereas the `is` check will keep working with subtypes.

Reply:
also I think the tween could be instantiated as `dynamic` and still used to tween a color.

Reply:
ah good call